src/averageAges.js

```diff
@@ -14,12 +14,18 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const centuriedMen = people
+    .filter(person => person.sex === 'm')
+    .filter(man => !century || Math.ceil(man.died / 100) === century);
```

Comment:
instead of iterating whole array twice add this condition to the first filter's callback (use logic operator to handle 2 conditions in the correct way)